lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
The original code just called standard active record api `find`.

If we want to deviate from that, would it make sense to just call `QueryRelation` with a block?

```ruby
QueryRelation.new(self, *args) do |mode, options|
  find(mode, to_legacy_options(options))
end
```

not sure if `QueryRelation` would need`self` other than looking up `klass`. Maybe passing that in would be better